packages/aws-cdk-lib/aws-s3/lib/notifications-resource/notifications-resource.ts

```diff
@@ -126,6 +127,13 @@ export class BucketNotifications extends Construct {
         }));
       }
 
+      // The customer resource handles unmanaged bucket is the right way so setting the managed flag to false
+      // Ading a feature flag to prevent it brings unexpected changes to customers
+      // Put it here because we still need to create the permission if it's unmanaged bucket.
+      if (cdk.FeatureFlags.of(this).isEnabled(cxapi.S3_KEEP_NOTIFICATION_IN_IMPORTED_BUCKET)) {
+        managed = false;
+      }
```

Comment:
I still don't get why we put it after the `addToRolePolicy` statement. If the variable `managed` is originally `false` but feature flag is on, then we treated it as unmanaged. It will go through the unmanaged code which will need the role policy, but it's not added.

Reply:
Updated.